File: scratch/inject_snt.py
import json, codecs

with open('c:/Users/EMRE/Desktop/Sihirbaz/snt_data.json', 'r', encoding='utf-8') as f:
    snt = json.load(f)

# 1. Update courses.js
with codecs.open('c:/Users/EMRE/Desktop/Sihirbaz/src/data/courses.js', 'r', 'utf-8') as f:
    c = f.read()

c = c.replace("'İTB Seçmeli'", "'ITB Seçmeli'")

insertIdx = c.rfind('];')
newCourses = ''
for crs in snt['courses']:
    if crs['id'] not in c:
        newCourses += f"  {{ id: '{crs['id']}', code: '{crs['code']}', name: '{crs['name'].replace(chr(39), chr(92)+chr(39))}', semester: 0, credit: {crs['credit']}, ects: {crs['ects']}, type: COURSE_TYPES.ITB, department: 'SNT', rawPrereq: '' }},\n"

c = c[:insertIdx] + newCourses + c[insertIdx:]
with codecs.open('c:/Users/EMRE/Desktop/Sihirbaz/src/data/courses.js', 'w', 'utf-8') as f:
    f.write(c)

# 2. Update schedule.js
with codecs.open('c:/Users/EMRE/Desktop/Sihirbaz/src/data/schedule.js', 'r', 'utf-8') as f:
    s = f.read()

sInsertIdx = s.rfind('];')
newSchedules = ''
for sch in snt['schedules']:
    if sch['crn'] not in s:
        newSchedules += f"  {{ crn: '{sch['crn']}', courseCode: '{sch['courseCode']}', courseId: '{sch['courseId']}', courseName: '{sch['courseName'].replace(chr(39), chr(92)+chr(39))}', day: '{sch['day']}', timeStart: '{sch['timeStart']}', timeEnd: '{sch['timeEnd']}', building: '{sch['building']}', classroom: '{sch['classroom']}', instructor: '{sch['instructor'].replace(chr(39), chr(92)+chr(39))}', capacity: {sch['capacity']}, enrolled: {sch['enrolled']} }},\n"

s = s[:sInsertIdx] + newSchedules + s[sInsertIdx:]
with codecs.open('c:/Users/EMRE/Desktop/Sihirbaz/src/data/schedule.js', 'w', 'utf-8') as f:
    f.write(s)

print("Courses and schedules updated.")

File: scratch/inject_snt.js
const fs = require('fs');

const snt = JSON.parse(fs.readFileSync('c:/Users/EMRE/Desktop/Sihirbaz/snt_data.json', 'utf8'));

// 1. Update courses.js
let c = fs.readFileSync('c:/Users/EMRE/Desktop/Sihirbaz/src/data/courses.js', 'utf8');

// Change İTB to ITB in COURSE_TYPE_LABELS
c = c.replace(/'İTB Seçmeli'/g, "'ITB Seçmeli'");

// Inject courses
const insertIdx = c.lastIndexOf('];');
let newCourses = '';
snt.courses.forEach(crs => {
    // If it's not already in courses.js
    if (!c.includes(crs.id)) {
        newCourses += `  { id: '${crs.id}', code: '${crs.code}', name: '${crs.name.replace(/'/g, "\\'")}', semester: 0, credit: ${crs.credit}, ects: ${crs.ects}, type: COURSE_TYPES.ITB, department: 'SNT', rawPrereq: '' },\n`;
    }
});
c = c.slice(0, insertIdx) + newCourses + c.slice(insertIdx);

fs.writeFileSync('c:/Users/EMRE/Desktop/Sihirbaz/src/data/courses.js', c);

// 2. Update schedule.js
let s = fs.readFileSync('c:/Users/EMRE/Desktop/Sihirbaz/src/data/schedule.js', 'utf8');
const sInsertIdx = s.lastIndexOf('];');
let newSchedules = '';
snt.schedules.forEach(sch => {
    if (!s.includes(sch.crn)) {
        newSchedules += `  { crn: '${sch.crn}', courseCode: '${sch.courseCode}', courseId: '${sch.courseId}', courseName: '${sch.courseName.replace(/'/g, "\\'")}', day: '${sch.day}', timeStart: '${sch.timeStart}', timeEnd: '${sch.timeEnd}', building: '${sch.building}', classroom: '${sch.classroom}', instructor: '${sch.instructor.replace(/'/g, "\\'")}', capacity: ${sch.capacity}, enrolled: ${sch.enrolled} },\n`;
    }
});
s = s.slice(0, sInsertIdx) + newSchedules + s.slice(sInsertIdx);
fs.writeFileSync('c:/Users/EMRE/Desktop/Sihirbaz/src/data/schedule.js', s);

console.log('Courses and schedules updated.');
